### PR TITLE
Is it a flashlight?

### PR DESCRIPTION
**Pull request by VALERA771**

### CODE REVIEW

**Comment by louis1706** [comment]
i don't thinks it's needed there no difference between FlashLight and Lantern

**Comment by VALERA771** [comment]
Yeah, but it's 2 separate items. I think we can create just a base class from which inhernits both flashlight and lantern

File: Exiled.API/Features/Items/Flashlight.cs [hunk 1]
```diff
@@ -54,18 +56,25 @@ public bool Active
...
+            NextAllowedTime = NextAllowedTime
         };
 
         /// <summary>
-        /// Returns the Flashlight in a human readable format.
+        /// Returns the Lantern in a human readable format.
         /// </summary>
-        /// <returns>A string containing Flashlight-related data.</returns>
-        public override string ToString() => $"{Type} ({Serial}) [{Weight}] *{Scale}* |{Active}|";
+        /// <returns>A string containing Lantern-related data.</returns>
```
Comment (louis1706) [inline Exiled.API/Features/Items/Flashlight.cs:77]: not a lantern but can be both

**Review by louis1706 — CHANGES_REQUESTED** [review]

**Comment by louis1706** [comment]
> Yeah, but it's 2 separate items. I think we can create just a base class from which inhernits both flashlight and lantern

yeah that what i was thinking

File: Exiled.API/Features/Items/Flashlight.cs [hunk 2]
```diff
@@ -54,18 +56,25 @@ public bool Active
...
+        /// <inheritdoc/>
+        public override Item Clone() => new Flashlight
+        {
+            IsEmittingLight = IsEmittingLight,
+            NextAllowedTime = NextAllowedTime
         };
 
         /// <summary>
-        /// Returns the Flashlight in a human readable format.
+        /// Returns the Lantern in a human readable format.
```
Comment (louis1706) [inline Exiled.API/Features/Items/Flashlight.cs:75]: same for this i would this you would seen it sorry

**Comment by VALERA771** [comment]
Breaking changes are allowed? Cuz I'm changing Base property type

**Comment by louis1706** [comment]
> Breaking changes are allowed? Cuz I'm changing Base property type

it's not a breaking change since it's already change inside Halloween also it's Base propperty it's should't break too mush

File: Exiled.API/Features/Items/Flashlight.cs [hunk 3]
```diff
@@ -31,19 +32,20 @@ public Flashlight(FlashlightItem itemBase)
...
         /// </summary>
-        public new FlashlightItem Base { get; }
+        /// <remarks>Can be <see cref="FlashlightItem"/> or <see cref="LanternItem"/>.</remarks>
+        public new ToggleableLightItemBase Base { get; }
 
         /// <summary>
-        /// Gets or sets a value indicating whether the flashlight is turned on.
+        /// Gets or sets a value indicating whether the item is emitting light.
         /// </summary>
-        public bool Active
```
Comment (louis1706) [inline Exiled.API/Features/Items/Flashlight.cs:46]: don't do this breaking change is not needed now

Comment (louis1706) [reply Exiled.API/Features/Items/Flashlight.cs:46]: you could make it Obsolete if you want

File: Exiled.API/Features/Items/Flashlight.cs [hunk 4]
```diff
@@ -31,19 +34,28 @@ public Flashlight(FlashlightItem itemBase)
         /// Initializes a new instance of the <see cref="Flashlight"/> class, as well as a new Flashlight item.
         /// </summary>
         internal Flashlight()
```
Comment (louis1706) [inline Exiled.API/Features/Items/Flashlight.cs:36]: add ItemType item argument to choose wish Item will be initialised

**Comment by VALERA771** [comment]
build go brrr

**Review by louis1706 — APPROVED** [review]

**Pull request merged**